File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        
        int fin_max = 0;
        unordered_map<int, bool>map;
        
        for(int i = 0; i < nums.size(); i++) {
            map[nums[i]] = true;
        }
        
        for(int i = 0; i < nums.size(); i++) {
            int temp_max = 1;
            if(map[nums[i] -1] != true) {
            while(map[nums[i]+temp_max] == true) {
                temp_max++;
            }
            }
            fin_max = max(temp_max, fin_max);
        }
        return fin_max;
    }
};

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        
        my_set = set(nums)
        max_seq = 0
            
        for num in my_set:
            if num-1 not in my_set:
                count = 1
                while num+count in my_set:
                    count += 1
                max_seq = max(count, max_seq)  
        return max_seq
            